### PR TITLE
Refactor of HDCamera (accesibility mostly) [Skip CI]

### PR DESCRIPTION
**Pull request by JulienIgnace-Unity**
### Purpose of this PR
The goal of this PR is to make everything that should not have been public, either internal or private and thus define a clearer public API for HDCamera
- Made internal/private APIs that should not have been public
- Fixed naming convention for private members
- Moved everything by accessibility for clarity (public, internal, then private)

Note to reviewer:
The goal is to assess whether or not APIs I made internal should stay public.
The question we need to ask is "Does a user writing code *outside* of HDRP need APIs XXX in a relevant way?".
Don't forget that for now, HDRP extensibility is almost non existent beside Custom Passes, Custom Post processes and Custom Render so most APIs are indeed internal (for example, Volumetric parameters can only be used inside HDRP and don' t make sense exposed to the users).

---
### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Camera/HDCamera.cs [hunk 1]
```diff
@@ -15,6 +15,7 @@ namespace UnityEngine.Rendering.HighDefinition
     // not used during a frame.
     public class HDCamera
     {
+        #region Public API
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Camera/HDCamera.cs]: as a reminder, Region aren't in our code convention chart. (I know there is some in our package, still want to note it)

**Comment by glabute** [comment]
Cinemachine is using HDCamera like this, please don't make it break:

#if CINEMACHINE_HDRP_7_0_0
            // Reset temporal effects
            var cam = brain.OutputCamera;
            if (cam != null)
            {
    #if CINEMACHINE_HDRP_7_1_0
                HDCamera hdCam = HDCamera.GetOrCreate(cam);
    #else
                HDCamera hdCam = HDCamera.GetOrCreate(cam, new XRPass());
    #endif
                hdCam.volumetricHistoryIsValid = false;
                hdCam.colorPyramidHistoryIsValid = false;
                hdCam.Reset();
            }
#endif

**Comment by Chman** [comment]
Hmm both `volumetricHistoryIsValid` and `colorPyramidHistoryIsValid` have been moved to the internal namespace. @JulienIgnace-Unity these two fields should probably be part of `Reset()` with all the other history resets.

**Comment by glabute** [comment]
I agree that Reset() should handle that, but if you change the public API you will break Cinemachine, so please leave it.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Camera/HDCamera.cs (hunk 1 above)
Comment (JulienIgnace-Unity) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Camera/HDCamera.cs]: Yes but honestly without that it's really hard to navigate in there.

**Pull request merged**